File: 프로그래머스/3/43164. 여행경로/여행경로.js
function solution(tickets) {
    const answer = [];
    const len = tickets.length;
    const visited = Array(len).fill(false);
    
    const dfs = (routes) => {
        if (routes.length === len+1) answer.push(routes);
        
        for (let i=0; i<len; i++) {
            const [start, end] = tickets[i];
            if (!visited[i] && routes.at(-1)===start) {
                visited[i] = true;
                dfs([...routes, end]);
                visited[i] = false;
            }
        }
    };
    
    dfs(["ICN"]);
    return answer.sort()[0];
}

File: 프로그래머스/3/43164. 여행경로/여행경로.py
def dfs(tickets, start, route):
    global visited, answer
    if 0 not in visited:
        answer.append(route)
    for i in range(len(tickets)):
        if tickets[i][0]==start and visited[i]==0:
            visited[i]=1
            dfs(tickets, tickets[i][1], [*route, tickets[i][1]])
            visited[i]=0
    
def solution(tickets):
    global visited, answer
    answer = []
    visited = [0 for _ in range(len(tickets))]
    stack = []
    dfs(tickets, "ICN", ["ICN"])
    return sorted(answer)[0]